server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -444,18 +433,7 @@ public ObjectMapper getObjectMapper(String name) {
      *                                  directly associated index-time analyzer
      */
     public NamedAnalyzer indexAnalyzer(String field, Function<String, NamedAnalyzer> unindexedFieldAnalyzer) {
```

Comment:
I remembered that search analyzers can be [reloaded](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-reload-analyzers.html) which doesn't cause a mapping update but changes how fields are analyzed! This is also an important way in which `MappingLookup` is not immutable.

Reply:
Yikes! We probably should figure out a way to make that replace the `MappingLookup` somehow. Do you want to save that for a follow up or for me to grab it now?

Reply:
These are required for `QueryShardContext` to get the snapshot, yeah.

Reply:
I think if we don't use the snapshot like we are here there are (rare) cases where you could see the addition of a new field without it getting in the snapshot's key. That is probably safe but its the kind of thing I don't want to have to think about and moving it over feels pretty safe too. But I think you are right about looking at this method again to make sure we actually need it.

Reply:
I think that in the near term we should still refactor `QueryShardContext` to not use `MapperService` to retrieve index analyzers. For me it makes the approach simpler and easier to understand. I agree though that this doesn't fix the search analyzer problem tracked in #66722, which I don't feel is super high priority and could be addressed at a later time.